### DISCUSSION TITLE
Website running Next.js with appDir, client components and SSG returns plain text response

### DISCUSSION THREAD

**Discussion by akoskm** [vercel/next.js#56471] [Help] [answered]
### Summary

I have a website deployed at DigitalOcean, with the default commands building the production version of the site:

Build Command
`npm run build`

Run Command
`npm start`

The following behavior happens randomly, and it makes the site unreliable for any kind of demo or presentation. This is a long standing issue. The first version of this was built in April when Next.js 13 and the appDir was pretty fresh. I was hoping that with time this will be fixed, but we're still experiencing the same behavior.

Does anyone know if this is caused by a plugin we're using?

![image](https://github.com/vercel/next.js/assets/3111627/70bdab7a-649c-4c86-a59a-1dfd43c27526)

### Additional information

```js
Here is our `package.json`:


{
  "name": "go",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "@graphql-codegen/cli": "^4.0.0",
    "@graphql-codegen/client-preset": "^4.0.0",
    "@types/node": "20.2.4",
    "@types/react": "18.2.7",
    "autoprefixer": "^10.4.14",
    "encoding": "^0.1.13",
    "eslint": "^8.41.0",
    "eslint-config-next": "^13.4.4",
    "graphql": "^16.6.0",
    "graphql-request": "^6.1.0",
    "marked": "^5.0.4",
    "node-loader": "^2.0.0",
    "postcss": "^8.4.23",
    "prettier": "^2.8.8",
    "prettier-plugin-tailwindcss": "^0.3.0",
    "tailwindcss": "^3.1.8",
    "typescript": "5.0.4"
  },
  "dependencies": {
    "@amplitude/analytics-browser": "^2.1.2",
    "discord.js": "^14.11.0",
    "next": "^13.5.4",
    "next-themes": "^0.2.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "sharp": "^0.32.3",
    "tiny-slider": "^2.9.4",
    "wowjs": "^1.1.3",
    "zlib-sync": "^0.1.8"
  }
}
```

And `next.config.js`:

```
/** @type {import('next').NextConfig} */
const nextConfig = {
  distDir: "_static",
  images: {
    unoptimized: true,
    remotePatterns: [
      {
        protocol: "https",
        hostname: "strapi.innotek.dev",
        port: "",
        pathname: "/uploads/**",
      },
    ],
  },
  webpack: (config, { isServer }) => {
    config.module.rules.push({
      test: /\.node$/,
      loader: "node-loader",
    });
    return config;
  },
};

module.exports = nextConfig;
```

**Comment by icyJoseph** [comment]
I think this is rooted on some server configuration. 

- https://github.com/vercel/next.js/issues/48569#issuecomment-1518887170
- https://github.com/vercel/next.js/issues/49130

**Reply by akoskm** [reply]
Thanks! It looks like this is an open issue.

In https://github.com/vercel/next.js/issues/49130 the author didn't confirm that it is fixed, and I'm encountering the same issue as well as the last commenter (whose comment was unfortunately marked as spam).

Next time this occurs - should be fairly easy to reproduce since it happens for certain URLs after each deployment - I'll send a screenshot showing the request/response headers.

**Reply by akoskm** [reply, marked as answer]
I realized I still had `"next": "13.4.4"` in my `package-lock.json` file, and after saving it with `13.5.4` - same as in `package.json` - the error didn't appear.

I pushed a few updates since this change, but the error didn't appear.

I'll close this discussion because, most likely, it was due to the older version in `package-lock.json`.

**Reply by s0h311** [reply]
After updating the packages in `package.json` manually, you should always reinstall the depedencies.

**Reply by akoskm** [reply]
I did. The problem was that I forgot to commit `package-lock.json`, which the DigitalOcean build uses to install production dependencies.
